### PR TITLE
[ADD] new module cron_inactivity_period

### PR DESCRIPTION
**Pull request by legalsylvain**
Add a new module ``cron_inactivity_period`` to allow users to configure period when the cron job will not be executed.

ir_cron_form

For the time being, periods are defined with a begin hour and an end hour. The module is designed to allow future developpement to add the possibility to define period with week days, or begin date / end date.

Note: a log is done if the cron is skipped.

review welcome. 

kind regards.

### CODE REVIEW

File: cron_inactivity_period/i18n/fr.po
Comment (fcayre) [inline cron_inactivity_period/i18n/fr.po]: Missing translation message

File: cron_inactivity_period/i18n/fr.po
Comment (fcayre) [inline cron_inactivity_period/i18n/fr.po]: Same thing for last 3 messages

File: cron_inactivity_period/i18n/fr.po
Comment (fcayre) [inline cron_inactivity_period/i18n/fr.po]: Typo: missing space before colon + inactivity -> inactivité

File: cron_inactivity_period/models/ir_cron.py
Comment (fcayre) [inline cron_inactivity_period/models/ir_cron.py:22]: Sorry to ask here, but doesn't Odoo use the classic "_" prefix meaning that this method is not public and subject to change? In which case overriding would be dangerous and to be carefully tested (which it does not seem to be).

**Review by fcayre — COMMENTED** [review]
A unittest with a call from the original cron model to your "_callback" override would be really helpful IMHO.

File: cron_inactivity_period/models/ir_cron.py
Comment (legalsylvain) [reply cron_inactivity_period/models/ir_cron.py:22]: > Sorry to ask here, but doesn't Odoo use the classic "_" prefix meaning that this method is not public 

Yes, this function will not be available via the API. (xml rpc call, etc...)

> and subject to change? 

No specially. And there are no changes when a version is stable. (in 8.0, there is no chance that the function changes, (excepted bug fixes))

File: cron_inactivity_period/i18n/fr.po
Comment (legalsylvain) [reply cron_inactivity_period/i18n/fr.po]: Hi. display name and "last update on" ... are fields added automatically by Odoo in all new models. I don't think that adding translation is usefull. (I added however)

**Comment by legalsylvain** [comment]
Hi @fcayre. Thanks for your review.
Changes done. (update translation, adding test).

File: cron_inactivity_period/README.rst
Comment (elicoidal) [inline cron_inactivity_period/README.rst]: use enumerated list `#. `
no need blank line in the list

File: cron_inactivity_period/__openerp__.py
Comment (elicoidal) [inline cron_inactivity_period/__openerp__.py:9]: space

File: cron_inactivity_period/models/ir_cron.py
Comment (elicoidal) [inline cron_inactivity_period/models/ir_cron.py]: is it skipped or runned :wink: 
I would say simply: ´Job `%s` skipped during inactivity period´

**Review by elicoidal — APPROVED** [review]
LGTM

File: cron_inactivity_period/tests/test_module.py
Comment (fcayre) [inline cron_inactivity_period/tests/test_module.py]: Small typo here (cosmetic remark)

**Review by hparfr — APPROVED** [review]
Looks good to me.

File: cron_inactivity_period/models/ir_cron.py
Comment (legalsylvain) [reply cron_inactivity_period/models/ir_cron.py]: depends, if it's a quantum job ;-)
Indeed the message was quite cryptic. I changed it.
Thanks.

File: cron_inactivity_period/__openerp__.py
Comment (legalsylvain) [reply cron_inactivity_period/__openerp__.py:9]: no. Not a good idea to add space ! it had broken Travis.
https://travis-ci.org/OCA/server-tools/jobs/420041893#L1105
regards.

**Review by andhit-r — APPROVED** [review]
LGTM

**Comment by andhit-r** [comment]
@OCA/server-environment-maintainers i think this PR can be merged since it got all the approval needed. Thanks.

**Pull request merged**
